### PR TITLE
Changes the RCD to only work on station levels

### PR DESCRIPTION
**Pull request by Arrow768**

### CODE REVIEW

**Comment by OneOneThreeEight** [comment]
Why punish people for taking the RCD, which is notorious for being an ammo consuming monster?

**Comment by Arrow768** [comment]
Because it is being used to break into areas they should not be able to get into on centcom and it can be used to simmilar effect on away missions / antag bases.

File: code/game/objects/items/weapons/RCD.dm [hunk 1]
```diff
@@ -71,6 +71,9 @@
 		return 0
 	if(istype(get_area(A),/area/shuttle)||istype(get_area(A),/turf/space/transit))
 		return 0
+	var/turf/t = get_turf(A)
+	if !(A.z in current_map.station_levels)
```
Comment (skull132) [inline code/game/objects/items/weapons/RCD.dm]:
```suggestion
	if (!(A.z in current_map.station_levels))
```

File: code/game/mecha/equipment/tools/tools.dm [hunk 2]
```diff
@@ -319,12 +319,15 @@
 	var/disabled = 0 //malf
 
 /obj/item/mecha_parts/mecha_equipment/tool/rcd/action(atom/target)
+	var/turf/t = get_turf(target)
 	if(istype(target,/area/shuttle)||istype(target, /turf/space/transit))//>implying these are ever made -Sieve
 		disabled = 1
+	else if (!isStationLevel(t.z))
```
Comment (Karolis2011) [inline code/game/mecha/equipment/tools/tools.dm]:
```suggestion
	else if (isNotStationLevel(t.z))
```

File: code/game/objects/items/weapons/RCD.dm [hunk 3]
```diff
@@ -71,6 +71,9 @@
 		return 0
 	if(istype(get_area(A),/area/shuttle)||istype(get_area(A),/turf/space/transit))
 		return 0
+	var/turf/t = get_turf(A)
+	if (!isStationLevel(t.z))
```
Comment (Karolis2011) [inline code/game/objects/items/weapons/RCD.dm]:
```suggestion
	if (isNotStationLevel(t.z))
```

**Comment by Arrow768** [comment]
Yes, it also effects the derelict.

**Pull request merged**